### PR TITLE
Add ability to write diagnostic to file.

### PR DESCRIPTION
**Pull request by AaronDonahue**
This commit will make it possible to request a diagnostic as output to file.  Each new diagnostic needs to be added to the register_diagnostics routine in order to be available to the scorpio_output class.  Requesting a diagnostic as output is as simple as adding the name that the diagnostic was registered under in the scorpio output YAML file.

An example can be found in the io_test which has been changed to also test for diagnostic output.

### CODE REVIEW

**Review by bartgol — CHANGES_REQUESTED** [review]
Looks good. I only have a few minor comments.

File: components/scream/src/diagnostics/potential_temperature.cpp [hunk 1]
```diff
@@ -45,6 +47,9 @@ void PotentialTemperatureDiagnostic::initialize_impl(const RunType /* run_type *
 
   const auto& output         = m_diagnostic_output.get_view<Pack**>();
 
+  auto ts = timestamp();  // TODO - how confident are we with how this timestamp is being handled?
```
[the remark this answers is not in the document]
Comment (AaronDonahue) [reply components/scream/src/diagnostics/potential_temperature.cpp:50]: okay good.  I just want to make sure that the timestamp in a diagnostic isn't necessarily out of sync with the rest of the model.  So I put this comment in the code to elicit just this kind of response.

**Comment by AaronDonahue** [comment]
@bartgol , thank you for the review.  I've made all the changes you asked for, when you get a chance can you re-review?

**Review by bartgol — APPROVED** [review]

**Comment by AaronDonahue** [comment]
This is failing because it needs a rebase for v0 baselines to pass and because of the fail we already know about for v1 tests that is fixed in PR #1512 .  Once #1512 goes in I'll rebase this PR and RETEST

**Review by bartgol — APPROVED** [review]
Looks good. I have a comment, but it's not a road blocker.

**Comment by AaronDonahue** [comment]
Retesting this PR, the only fails were `blake` which is expected and the one fail on `mappy` was the `p3_tests_omp13` which shouldn't be impacted by this PR at all.

**Comment by AaronDonahue** [comment]
adding the `skip blake` label so the AT won't time-out like above.

**Comment by AaronDonahue** [comment]
Luca, this passed on `mappy` and `weaver`.  I'm going to manually run on `quartz`.  In the meantime do you mind rereviewing?

**Comment by AaronDonahue** [comment]
Testing on quartz returned a PASS for all of the test suites.

**Review by bartgol — APPROVED** [review]

**Pull request merged**